perun-core/src/main/java/cz/metacentrum/perun/core/impl/GroupsManagerImpl.java

```diff
@@ -111,6 +114,20 @@
 		return pair;
 	};
 
+	private static final ResultSetExtractor<Map<Member, List<Integer>>> MEMBER_MEMBERGROUPSTATUS_EXTRACTOR = resultSet -> {
+		Map<Member, List<Integer>> map = new HashMap<>();
+		while (resultSet.next()) {
+			Member member = MembersManagerImpl.MEMBER_MAPPER.mapRow(resultSet, resultSet.getRow());
+			List<Integer> list = new ArrayList<>();
+			if (map.get(member) != null) {
+				list = map.get(member);
+			}
```

Comment:
This might help improve performance.
```suggestion
			List<Integer> list = list = map.get(member);
			if (list == null) {
			    list = new ArrayList<>();
			}
```

Reply:
Fixed.